sync-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -74,6 +77,15 @@
 
     public final HashMap<String, String> requestProperties;
 
+    public final LinkedList<HttpConnectionRequestFilter> requestFilters;
+    public final LinkedList<HttpConnectionResponseFilter> responseFilters;
+
+    //we need to store the response code separately so we don't send a request
+    //before we are ready
+    private int responseCode = -1;
```

Comment:
This still feels messy, and isn't publicly documented that -1 can be returned by getResponseCode.